toy_robot.rb

```diff
@@ -7,34 +7,47 @@ def initialize
     @x = 0
     @y = 0
     @facing = :north
+    @movements = []
   end
 
   def place(x,y,facing)
     raise InvalidPositionError.new("Invalid Position, #{x}, #{y}, #{facing}") if !(x.between?(0,4)) || !(y.between?(0,4))
     @x = x
     @y = y
     @facing = facing
+    @movements.replace(["PLACE #{@x}, #{@y}, #{@facing.to_s.upcase}"])
   end
 
   def move
```

Comment:
As a tip, for the refactor, you should probably separate the code for validating and moving. Try raising an exception if the the move is invalid. Then setting co-ordinates and movements if valid.